adapter/adapter-webserver/src/main/java/com/dtp/adapter/webserver/TomcatDtpAdapter.java

```diff
@@ -60,54 +59,99 @@ public ThreadPoolStats getPoolStats() {
 
     @Override
     public void refresh(DtpProperties dtpProperties) {
-        TpExecutorProps props = dtpProperties.getTomcatTp();
-        if (Objects.isNull(props) || containsInvalidParams(props, log)) {
-            return;
+        refresh(POOL_NAME, executorWrapper, dtpProperties.getPlatforms(), dtpProperties.getTomcatTp());
+    }
+    
+    /**
+     * TomcatExecutorAdapter implements ExecutorAdapter, the goal of this class
+     * is to be compatible with {@link org.apache.tomcat.util.threads.ThreadPoolExecutor}.
+     **/
+    private static class TomcatExecutorAdapter implements ExecutorAdapter<ThreadPoolExecutor> {
```

Comment:
Can ThreadPoolExecutorAdapter be used here?

Reply:
> Can ThreadPoolExecutorAdapter be used here?

No, `ThreadPoolExecutorAdapter` is for `java.util.concurrent.ThreadPoolExecutor`, and the `TomcatExecutorAdapter` is for `org.apache.tomcat.util.threads.ThreadPoolExecutor`, although the implementation is similar, it is necessary.

Reply:
that's right